### PR TITLE
add msgStringOverrides, skin.setSpritePrefix

### PR DESCRIPTION
**Pull request by cpirich**
- allow a level to re-map any string by populating `msgStringOverrides`
- allow a skin to set the prefix for the `setSprite` block by defining `skin.setSpritePrefix` (typically defined to `msg.setSprite()`) - NOTE: this only applies to `setSprite`, not `setSpriteParams` and all of the other variations.
- use `msgStringOverrides` to make blockly level 9 show `whenGetCharacterRebelPilot` instead of `whenTouchGoal`
- update hoc2015 script to leverage `skin.setSpritePrefix` to it says `set droid to <>`

### CODE REVIEW

**Comment by cpirich** [comment]
@breville @islemaster

File: apps/src/studio/skins.js [hunk 1]
```diff
@@ -923,9 +925,11 @@ function loadHoc2015x(skin, assetUrl) {
     ];
 
   skin.spriteChoices = [
-    [msg.setSpriteHidden(), HIDDEN_VALUE],
-    [msg.setSpriteRandom(), RANDOM_VALUE],
-    [msg.setSpriteBB8(), '"BB-8"']];
+    [msg.setDroidHidden(), HIDDEN_VALUE],
+    [msg.setDroidRandom(), RANDOM_VALUE],
+    [msg.setDroidBB8(), '"BB-8"']];
```
Comment (islemaster) [inline apps/src/studio/skins.js]: Should this be lowercase, since we use "r2-d2" and "c-3po" above?

**Comment by islemaster** [comment]
LGTM

File: apps/src/studio/skins.js (hunk 1 above)
Comment (cpirich) [inline apps/src/studio/skins.js]: yes, this block doesn't actually work for this skin, but we should set it that way so that it could.

**Pull request merged**